### PR TITLE
Major full repo cleanup and lots of  bug fixes

### PR DESCRIPTION
**Pull request by jack60612**
Fixes a couple of warnings but mostly just updates a lot of packages to the latest compatible version.
Some unused packages were removed.

### CODE REVIEW

**Comment by jack60612** [comment]
All warnings on main branch were fixed but the farmer view is still broken

**Comment by jack60612** [comment]
Api calls need to be updated

**Comment by jack60612** [comment]
I'm done for now so feel free to give me tips

File: src/components/core/components/Card/Card.tsx [hunk 1]
```diff
@@ -50,7 +50,7 @@ export default function Card(props: Props) {
   }
 
   return (
-    <StyledCardMaterial onClick={handleClick} clickable={!!onSelect} disabled={disabled}>
+    <StyledCardMaterial onClick={handleClick} $clickable={!!onSelect} disabled={disabled}>
```
Comment (seeden) [inline src/components/core/components/Card/Card.tsx:53]: $clickable property does not exists

File: src/components/core/components/CardKeyValue/CardKeyValue.tsx [hunk 2]
```diff
@@ -39,14 +39,14 @@ export default function CardKeyValue(props: Props) {
       <TableBody>
         {rows.map((row) => (
           <TableRow key={row.key}>
-            <StyledTableCell hideDivider={hideDivider}>
-              <Typography variant="body1" color="textSecondary" noWrap>
+            <StyledTableCell $hideDivider={hideDivider}>
```
Comment (seeden) [inline src/components/core/components/CardKeyValue/CardKeyValue.tsx:42]: $hideDivider property does not exists

File: src/components/core/components/Card/Card.tsx (hunk 1 above)
Comment (jack60612) [reply src/components/core/components/Card/Card.tsx:53]: the $ makes it so that the react dom does not yell at us about it

File: src/components/core/components/CardKeyValue/CardKeyValue.tsx (hunk 2 above)
Comment (jack60612) [reply src/components/core/components/CardKeyValue/CardKeyValue.tsx:42]: the $ makes it so that the react dom does not yell at us about it

File: src/components/core/components/CardKeyValue/CardKeyValue.tsx [hunk 3]
```diff
@@ -39,14 +39,14 @@ export default function CardKeyValue(props: Props) {
...
+              <Typography component='div' variant="body1" color="textSecondary" noWrap>
```
Comment (seeden) [inline src/components/core/components/CardKeyValue/CardKeyValue.tsx:43]: Why are you overriding p tag with div?

Comment (jack60612) [reply src/components/core/components/CardKeyValue/CardKeyValue.tsx:43]: you dont want p's to be in other p's or other div as it will give a dom warning

File: src/components/farm/Farm.tsx [hunk 4]
```diff
@@ -28,7 +28,7 @@ export default function Farm() {
           <>
             <FarmLastAttemptedProof />
             <FarmLatestBlockChallenges />
-            <AdvancedOptions>
+            <AdvancedOptions expanded>
```
Comment (seeden) [inline src/components/farm/Farm.tsx:31]: advanced options are collapsed by default

File: src/components/core/components/CardKeyValue/CardKeyValue.tsx (hunk 2 above)
Comment (jack60612) [reply src/components/core/components/CardKeyValue/CardKeyValue.tsx:42]: react dom as i said before

File: src/components/farm/FarmLatestBlockChallenges.tsx [hunk 5]
```diff
@@ -67,6 +67,7 @@ export default function FarmLatestBlockChallenges() {
             color="primary"
             href="https://www.chiaexplorer.com/"
             target="_blank"
+            to="/"
```
Comment (seeden) [inline src/components/farm/FarmLatestBlockChallenges.tsx:70]: we need to fix component prop types and to add unused property

File: src/components/farm/overview/FarmOverview.tsx [hunk 6]
```diff
@@ -37,7 +37,7 @@ export default function FarmOverview() {
             <Trans>Your Farm Overview</Trans>
           </Typography>
         </Flex>
-        <More>
+        <More open>
```
Comment (seeden) [inline src/components/farm/overview/FarmOverview.tsx:40]: more is not open by default

File: src/components/plot/add/PlotAdd.tsx [hunk 7]
```diff
@@ -36,7 +36,7 @@ export default function PlotAdd() {
   const fingerprint = useSelector(
     (state: RootState) => state.wallet_state.selected_fingerprint,
   );
-  const addNFTref = useRef();
+  const addNFTref = createRef();
```
Comment (seeden) [inline src/components/plot/add/PlotAdd.tsx:39]: functional components are not using createRef. correct hook use useRef

File: src/components/plot/add/PlotAddSelectFinalDirectory.tsx [hunk 8]
```diff
@@ -56,7 +56,7 @@ export default function PlotAddSelectFinalDirectory() {
           size="large"
           variant="outlined"
           selected={hasFinalLocation}
-          nowrap
+          nowrap={1}
```
Comment (seeden) [inline src/components/plot/add/PlotAddSelectFinalDirectory.tsx:59]: nowrap is boolean property

File: src/components/plotNFT/PlotExternalNFTCard.tsx [hunk 9]
```diff
@@ -282,7 +282,7 @@ export default function PlotExternalNFTCard(props: Props) {
 
           <Flex flexDirection="column" gap={2} flexGrow={1}>
             <Flex flexDirection="column" flexGrow={1}>
-              <CardKeyValue rows={rows} hideDivider />
+              <CardKeyValue rows={rows} $hideDivider />
```
Comment (seeden) [inline src/components/plotNFT/PlotExternalNFTCard.tsx:285]: $hideDivider does not exists

File: src/components/plotNFT/PlotNFTCard.tsx [hunk 10]
```diff
@@ -320,13 +320,13 @@ export default function PlotNFTCard(props: Props) {
...
             <StyledInvisibleContainer>
-              <Typography variant="body2" noWrap>
+              <Typography component='div' variant="body2" noWrap>
                 {!!pool_url && (
                   <Flex alignItems="center" gap={1}>
                     <Typography variant="body2" color="textSecondary">
                       <Trans>Pool:</Trans>
                     </Typography>
-                    <Link target="_blank" href={pool_url}>
+                    <Link target="_blank" href={pool_url} to='/'>
```
Comment (seeden) [inline src/components/plotNFT/PlotNFTCard.tsx:329]: use href or to. not both

File: src/components/plot/add/PlotAddSelectFinalDirectory.tsx (hunk 8 above)
Comment (jack60612) [reply src/components/plot/add/PlotAddSelectFinalDirectory.tsx:59]: 1 was added so that react would not output a warning saying it isn't a boolean.
1= true

File: src/components/pool/PoolOverview.tsx [hunk 11]
```diff
@@ -189,6 +189,7 @@ export default function PoolOverview() {
         </Flex>
         {showTable ? (
           <Table
+            component='div'
```
Comment (seeden) [inline src/components/pool/PoolOverview.tsx:192]: do not override table component

File: src/components/wallet/WalletImport.tsx [hunk 12]
```diff
@@ -53,7 +53,6 @@ function MnemonicField(props: TextFieldProps & MnemonicFieldProps) {
             label={label}
             error={error}
             autoFocus={autoFocus}
```
Comment (seeden) [inline src/components/wallet/WalletImport.tsx:55]: why did you remove default value?

File: src/components/plotNFT/PlotExternalNFTCard.tsx (hunk 9 above)
Comment (jack60612) [reply src/components/plotNFT/PlotExternalNFTCard.tsx:285]: let me find the documentation i was referring too

File: src/components/plotNFT/PlotNFTCard.tsx (hunk 10 above)
Comment (jack60612) [reply src/components/plotNFT/PlotNFTCard.tsx:329]: let me fix that Thanks!

File: src/components/core/components/Card/Card.tsx (hunk 1 above)
Comment (seeden) [reply src/components/core/components/Card/Card.tsx:53]: we can fix it directly in the styled component

File: src/components/core/components/CardKeyValue/CardKeyValue.tsx (hunk 2 above)
Comment (seeden) [reply src/components/core/components/CardKeyValue/CardKeyValue.tsx:42]: we can fix it directly in the styled component

File: src/components/wallet/WalletImport.tsx (hunk 12 above)
Comment (jack60612) [reply src/components/wallet/WalletImport.tsx:55]: This is from the warning it gave me "Input elements must be either controlled or uncontrolled"

**Review by seeden — CHANGES_REQUESTED** [review]
thank you, but we need to fix few things in a more proper way

File: src/components/plot/add/PlotAddSelectFinalDirectory.tsx (hunk 8 above)
Comment (seeden) [reply src/components/plot/add/PlotAddSelectFinalDirectory.tsx:59]: nowrap alone is boolean by default

File: src/components/pool/PoolOverview.tsx (hunk 11 above)
Comment (jack60612) [reply src/components/pool/PoolOverview.tsx:192]: let me remove that!

File: src/components/plotNFT/PlotNFTCard.tsx (hunk 10 above)
Comment (jack60612) [reply src/components/plotNFT/PlotNFTCard.tsx:329]: to is used to not give an error that to is missing but let me fix that for the link module

File: src/components/plot/add/PlotAddSelectFinalDirectory.tsx (hunk 8 above)
Comment (jack60612) [reply src/components/plot/add/PlotAddSelectFinalDirectory.tsx:59]: > nowrap alone is boolean by default

react itself as a bug right now where it will give a warning if you don't do that

File: src/components/farm/Farm.tsx (hunk 4 above)
Comment (jack60612) [reply src/components/farm/Farm.tsx:31]: let me fix that
my ide auto added that sorry.

File: src/components/farm/overview/FarmOverview.tsx (hunk 6 above)
Comment (jack60612) [reply src/components/farm/overview/FarmOverview.tsx:40]: thanks!

File: src/components/core/components/Card/Card.tsx (hunk 1 above)
Comment (jack60612) [reply src/components/core/components/Card/Card.tsx:53]: pushing these changes now but will do

Comment (jack60612) [reply src/components/core/components/Card/Card.tsx:53]: @seeden sorry about all of the eh changes.
this is my first time using react and typescript

Comment (jack60612) [reply src/components/core/components/Card/Card.tsx:53]: Just figured out how to fix it the right way

File: src/components/farm/FarmLatestBlockChallenges.tsx (hunk 5 above)
Comment (jack60612) [reply src/components/farm/FarmLatestBlockChallenges.tsx:70]: Could you lend me a hand with that?

File: src/components/plot/add/PlotAdd.tsx (hunk 7 above)
Comment (jack60612) [reply src/components/plot/add/PlotAdd.tsx:39]: Sorry

Comment (jack60612) [reply src/components/plot/add/PlotAdd.tsx:39]: Fixing

File: src/components/plot/add/PlotAddSelectFinalDirectory.tsx (hunk 8 above)
Comment (jack60612) [reply src/components/plot/add/PlotAddSelectFinalDirectory.tsx:59]: @seeden do you still want me to change it?

File: src/components/core/components/CardKeyValue/CardKeyValue.tsx (hunk 2 above)
Comment (jack60612) [reply src/components/core/components/CardKeyValue/CardKeyValue.tsx]: > we can fix it directly in the styled component

Thank you i fixed it

File: src/components/plot/add/PlotAddSelectFinalDirectory.tsx (hunk 8 above)
Comment (jack60612) [reply src/components/plot/add/PlotAddSelectFinalDirectory.tsx:59]: I might be able to get it now

Comment (jack60612) [reply src/components/plot/add/PlotAddSelectFinalDirectory.tsx:59]: Got it!

File: src/components/farm/FarmLatestBlockChallenges.tsx (hunk 5 above)
Comment (jack60612) [reply src/components/farm/FarmLatestBlockChallenges.tsx:70]: > Could you lend me a hand with that?

i think i got it

Comment (jack60612) [reply src/components/farm/FarmLatestBlockChallenges.tsx]: Let me know what you think

File: src/components/core/components/CardKeyValue/CardKeyValue.tsx [hunk 13]
```diff
@@ -60,6 +63,6 @@ export default function CardKeyValue(props: Props) {
 
 CardKeyValue.defaultProps = {
   label: undefined,
-  hideDivider: false,
+  hideDivider: 0,
```
Comment (seeden) [inline src/components/core/components/CardKeyValue/CardKeyValue.tsx:66]: hideDivider is boolean. please use correct false value

Comment (jack60612) [reply src/components/core/components/CardKeyValue/CardKeyValue.tsx:66]: Sorry about that.
Let me fix it

File: src/components/core/components/Link/Link.tsx [hunk 14]
```diff
@@ -20,8 +20,8 @@ type Props = BaseLinkProps &
     | RouterLinkProps
   );
 
-const StyledBaseLink = styled(({ fullWidth, noWrap, ...rest }) => (
-  <BaseLink {...rest} />
+const StyledBaseLink = styled(({ fullWidth, noWrap,to, ...rest }) => (
+  <BaseLink to={to ?? "/"} {...rest} />
```
Comment (seeden) [inline src/components/core/components/Link/Link.tsx:24]: the parameter "to" will not have default value "/". the correct default value is undefined. please remove this change completly

File: src/components/selectKey/SelectKey.tsx [hunk 15]
```diff
@@ -46,9 +46,9 @@ export default function SelectKey() {
   const hasFingerprints =
     publicKeyFingerprints && !!publicKeyFingerprints.length;
 
-  async function handleClick(fingerprint: Fingerprint) {
-    await dispatch(resetMnemonic());
-    await dispatch(login_action(fingerprint));
+  function handleClick(fingerprint: Fingerprint) {
```
Comment (seeden) [inline src/components/selectKey/SelectKey.tsx:49]: why did you remove async/await?

File: src/electron/main.tsx [hunk 16]
```diff
@@ -140,15 +140,17 @@ if (!handleSquirrelEvent()) {
       });
 
       if (dev_config.redux_tool) {
-        BrowserWindow.addDevToolsExtension(
-          path.join(os.homedir(), dev_config.redux_tool),
-        );
+        const reduxDevToolsPath = path.join(os.homedir(), dev_config.react_tool)
+        app.whenReady().then(async () => {
```
Comment (seeden) [inline src/electron/main.tsx:144]: you can use 

await app.whenReady();
await session.defaultSession.loadExtension(reduxDevToolsPath);

**Review by seeden — CHANGES_REQUESTED** [review]
Thank you for your update. Added few comments/questions

File: src/components/core/components/Link/Link.tsx (hunk 14 above)
Comment (jack60612) [reply src/components/core/components/Link/Link.tsx]: fixed

File: src/components/selectKey/SelectKey.tsx (hunk 15 above)
Comment (jack60612) [reply src/components/selectKey/SelectKey.tsx:49]: Its not an async/await function so it isn't needed

Comment (jack60612) [reply src/components/selectKey/SelectKey.tsx:49]: await does nothing in that case

File: src/electron/main.tsx (hunk 16 above)
Comment (jack60612) [reply src/electron/main.tsx]: fixed

**Comment by seeden** [comment]
Thank you, I think we can merge it tomorrow.

**Review by seeden — APPROVED** [review]

**Comment by jack60612** [comment]
> Thank you, I think we can merge it tomorrow.

Sorry about all the changes needed btw, As i said first time using react and jsx.
@seeden Thanks for everything!

**Comment by seeden** [comment]
Sorry for the delay. Still verifying few removed dependencies. It looks like a webpack-dev-server is required for the "dev" script

**Comment by jack60612** [comment]
> Sorry for the delay. Still verifying few removed dependencies. It looks like a webpack-dev-server is required for the "dev" script

I just reinstalled everything using npm ci and im able to successfully run `npm run dev:fast`
@seeden

**Comment by seeden** [comment]
[0] [webpack-cli] For using 'serve' command you need to install: 'webpack-dev-server' package
[0] [webpack-cli] Would you like to install 'webpack-dev-server' package? (That will run 'npm install -D webpack-dev-server

**Comment by seeden** [comment]
I already fixed it with another PR and also needed to downgrade one package related to localstorage because new versions are broken as well.

**Comment by jack60612** [comment]
> I already fixed it with another PR and also needed to downgrade one package related to localstorage because new versions are broken as well.

Sorry you had to do that.
Have a good day!

**Comment by seeden** [comment]
no problem :) you too
